packages/ui-lib/src/Misc/molecules/Pagination.tsx

```diff
@@ -0,0 +1,351 @@
+    }
+
+  }, [isValidInput, totalPages]);
+
+
+  const onBlur = useCallback(({ target: { value } }: React.FocusEvent<HTMLInputElement>) => {
+
+    if ((value === '')) {
+      setPageValue(activePage.toString());
+
+    } else if (isValidInput(value) && parseInt(value) !== activePage) {
+      setDisabledGo(false);
+      setFieldState('processing');
+      return;
+
+    } else if (!isValidInput(value)) {
+      setFieldState('invalid');
+      setDisabledGo(true);
+      return;
+    }
+
+    setFieldState('default');
+
+  }, [activePage, isValidInput]);
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+
+    if ((event.key === 'Enter') && isValidInput(pageValue)) {
+      onClickGo();
+    }
+
+  };
+
+  /**
+   * Review if current edith is valid if not make disable
+   */
+  const onFocus = ({ target: { value } }: React.FocusEvent<HTMLInputElement>) => {
+
+    if (isValidInput(value)) {
+      setFieldState('processing');
+    } else {
+      setFieldState('invalid');
+      setDisabledGo(true);
+    }
+
+  };
+
+  const getTextColor = useCallback((state: string): string => {
+
+    switch (state) {
+      case 'processing':
+        return 'var(--input-border-processing, #4db5ff)';
+        break;
+
+      case 'invalid':
+        return 'var(--input-border-invalid, #f66)';
+
+      case 'default':
+      default:
+        return 'var(--input-border-default, --grey-9)';
+    }
+
+  }, []);
+
+  const onClickGo = useCallback(() => {
+
+    onPageChange(parseInt(pageValue));
+    inputRef.current && inputRef.current.blur();
+    setDisabledGo(true);
+    setFieldState('default');
+
+  }, [onPageChange, pageValue]);
+
+  const handlePageChange = (value: number) => {
+    onPageChange(value);
+    setPageValue(value.toString());
+  };
+
+  const onItemsSelectChange = useCallback((value: string) => {
+    onItemsChange(Number(value));
+  }, [onItemsChange]);
+
+  const handlePaste = (e: React.ClipboardEvent<HTMLInputElement>) => {
+    const clipboardData = e.clipboardData || (window as any).clipboardData; // Cross-browser support
+    const pastedText = clipboardData.getData('text');
+
+    if (!/^\d+$/.test(pastedText)) {
+      e.preventDefault();
+    }
+  };
+
+  return (
+    <PaginationContainer>
+      <ItemsSelectWrapper width={selectWidth}>
+        <SelectField
+          disabled={selectDisabled}
+          label={{ htmlFor: 'paginationPages', text: itemsText, isSameRow: true }}
+          defaultValue={1}
+          changeCallback={onItemsSelectChange}
+          marginBottom={false}
+        >
+          <Fragment>
+            {itemsOptions.map(({ value, textValue }, index) =>
+              <option key={index} value={value}>{textValue}</option>
+            )}
+          </Fragment>
+        </SelectField>
+      </ItemsSelectWrapper>
+      <PageLabel htmlFor='goButton'>{pageText}</PageLabel>
+      <InputContainer borderColor={getTextColor(fieldState)} shouldShake={shouldShake}>
```

Comment:
This is using `getTextColor` for `borderColor` which is why I think we've got some mix ups that I've commented on elsewhere. I think we either need to separate them or replace them with something else.

Reply:
Agreed